pkg/certificate/providers/config.go

```diff
@@ -40,47 +42,65 @@ var getCA func(certificate.Issuer) (pem.RootCertificate, error) = func(i certifi
 }
 
 // NewCertificateManager returns a new certificate manager, with an MRC compat client.
-// TODO(4502): Use an informer behind a feature flag.
-func NewCertificateManager(kubeClient kubernetes.Interface, kubeConfig *rest.Config, cfg configurator.Configurator,
-	providerNamespace string, options Options, msgBroker *messaging.Broker) (*certificate.Manager, error) {
+// TODO(4713): Use an informer behind a feature flag.
+func NewCertificateManager(ctx context.Context, kubeClient kubernetes.Interface, kubeConfig *rest.Config, cfg configurator.Configurator,
+	providerNamespace string, options Options, msgBroker *messaging.Broker, informers *informers.InformerCollection) (*certificate.Manager, error) {
 	if err := options.Validate(); err != nil {
 		return nil, err
 	}
 
-	// TODO(4502): Switch the compat client to an informer. Might need another struct to compose the informer and
-	// provider generator.
-	mrcClient := &MRCCompatClient{
-		MRCProviderGenerator: MRCProviderGenerator{
-			kubeClient:      kubeClient,
-			kubeConfig:      kubeConfig,
-			KeyBitSize:      cfg.GetCertKeyBitSize(),
-			caExtractorFunc: getCA,
-		},
-		mrc: &v1alpha2.MeshRootCertificate{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      "legacy-compat",
-				Namespace: providerNamespace,
-				Annotations: map[string]string{
-					constants.MRCVersionAnnotation: "legacy-compat",
-				},
+	var mrcClient certificate.MRCClient
+	// TODO(4713): Switch out with feature flag or some other mechanism
+	if true {
```

Comment:
I'm not sure we need an extra feature flag toggle. I'd consider just having one or the other, and making the switch at some point in time.

Features are pinned to releases and too many feature flags conflates and confuses users with internal deatils they won't need IMO

Reply:
True...what I might do is default to MRCs, but if there are none, use the compat client